source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -177,6 +178,10 @@ void Filter::complete(Filters::Common::RateLimit::LimitStatus status,
       callbacks_->streamInfo().setResponseFlag(StreamInfo::ResponseFlag::RateLimitServiceError);
     }
   } else if (!initiating_call_) {
+    Http::HeaderMapPtr upstream_headers_to_add = std::move(upstream_headers);
```

Comment:
What about the other `continueDecoding()` case above?

Reply:
Made changes to forward request headers (if present) to upstream if failure mode is allowed.